pehlaschoollibrary/proj.android-studio/app/src/com/maq/pehlaschool/library/SplashScreenActivity.java

```diff
@@ -50,6 +48,15 @@
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        Intent localeIntent = getIntent();
+        Bundle extras = localeIntent.getExtras();
+        if (extras != null && extras.getString("locale") != null && !extras.getString("locale").isEmpty()) {
+            locale = extras.getString("locale").toLowerCase();
+        } else {
+            // set the default value of the variable on successive calls
+            locale = "english";
+        }
```

Comment:
Looks like this block of code is getting repeated at least twice.

Reply:
Added getLocalefromIntent() function to the SelectActivity and used it.